### ISSUE TITLE
add gpu statistics

### ISSUE THREAD

**Issue by yee379** [vpenso/prometheus-slurm-exporter#23]
it would be great if a count of gpus (and other gres's) could be provided in the metrics :)

**Comment by captainvera** [comment]
👆 +1 on this idea.

Monitoring of Gres resources is the n1 reason we were looking at slurm exporting to prometheus/graffana.

I'm not familiar with the codebase, how hard would this be to implement @mtds?

**Comment by captainvera** [comment]
Hey @mtds I'd like to tackle this and submit a PR. Any pointers on where to start?

**Comment by mtds** [comment]
I will most likely start to look into this issue soon but it's not yet on top of the list of my priorities.

We do have a small GPU cluster but the usage of this hardware is not tracked, mostly because the
cluster itself is still tied to a very old version of Slurm (and it will likely to be phased out soon).
So, at the moment I do not have much experience with newer versions of Slurm and GPUs. I know that once Slurm is configured to deal with this kind of hardware it should be possible to visualize how these resources are allocated.

If you are familiar with this exporter codebase, you have seen that it's nothing more than a Golang
wrapper on top of  ``squeue``, ``sinfo``, ``sdiag`` (plus the Golang client to interact with Prometheus) and quite a bit of output parsing.

While browsing the slurm-users mailing lists, two commands to start tinkering with would be the following:

* ``sinfo -o %G`` - gives total gpu in cluster
* ``squeue -o %b``   - gives amount of gpu in use for each running job

I would start to create a new Golang file (like ``nodes.go``) and proceed from there to parse the output provided by the Slurm commands.

Sorry, I realize that there's not so much as a suggestion but if you manage to make a PR I would be glad to evaluate it.

**Comment by captainvera** [comment]
Thanks for the quick overview on things!
I'll take a look at the existing code and I'll try to replicate it's structure in a `nodes` file. Internally we already aliased those commands to the normal `squeue` so we can visualize utilization so it shouldn't be too hard to parse that.

Overall doesn't look too difficult :) I'll make a PR once I'm happy with it!

**Comment by moinnadeem** [comment]
Let me know if you need a hand, I was wanting something like this and will be happy to help! @captainvera

**Comment by captainvera** [comment]
hey @moinnadeem, my company recently had some unexpected restructuring and this is no longer a priority. As such, I didn't even start working on it :/ 

Sorry I didn't update this issue. Even though I can't extend my help on this right now, I'll be very very excited if you do pick this up!

**Comment by JoeriHermans** [comment]
Pull request #34 might be relevant.

**Comment by mtds** [comment]
Contributed pull request from @JoeriHermans was finally merged! :-)
